app/actors/hyrax/actors/file_set_actor.rb

```diff
@@ -189,16 +214,61 @@ def perform_save(object)
+        def object_to_act_on(object, force_use_valkyrie = use_valkyrie)
           return object if valkyrie_object?(object)
-          use_valkyrie ? object.valkyrie_resource : object
+          force_use_valkyrie ? object.valkyrie_resource : object
+        end
+
+        # if passed an ActiveFedora object, return it; otherwise, convert resource to an ActiveFedora object and return
+        # TODO: Remove this method when env passes resources instead of active fedora objects
+        def convert_to_af_object(object)
+          return object if af_object?(object)
+          Wings::ActiveFedoraConverter.new(resource: object).convert
         end
 
         # determine if the object is a valkyrie resource
         # TODO: Remove this method when env passes resources instead of active fedora objects
         def valkyrie_object?(object)
           object.is_a? Valkyrie::Resource
         end
+
+        # determine if the object is an ActiveFedora object
+        # TODO: Remove this method when env passes resources instead of active fedora objects
+        def af_object?(object)
+          object.is_a? ActiveFedora::Base
+        end
+
+        # switches between using valkyrie to attach_to_work and using active fedora
+        # TODO: Remove this method when env passes resources instead of active fedora objects
+        def perform_attach_to_work(work, file_set_params)
+          force_use_valkyrie = use_valkyrie || valkyrie_object?(work) || valkyrie_object?(file_set)
+          if force_use_valkyrie
+            source_work = object_to_act_on(work, force_use_valkyrie)
+            target_file_set = object_to_act_on(file_set, force_use_valkyrie)
+            updated_work = valk_attach_to_work(source_work, target_file_set)
+            work_to_return = valkyrie_object?(work) ? updated_work : convert_to_af_object(updated_work)
+          else
+            work_to_return = af_attach_to_work(work, file_set_params)
+          end
+          work_to_return
+        end
+
+        # TODO: Remove this method when env passes resources instead of active fedora objects
+        def af_attach_to_work(work, file_set_params)
+          acquire_lock_for(work.id) do
+            # Ensure we have an up-to-date copy of the members association, so that we append to the end of the list.
+            work.reload unless work.new_record?
+            file_set.visibility = work.visibility unless assign_visibility?(file_set_params)
+            work.ordered_members << file_set
+            work.representative = file_set if work.representative_id.blank?
+            work.thumbnail = file_set if work.thumbnail_id.blank?
+            # Save the work so the association between the work and the file_set is persisted (head_id)
+            # NOTE: the work may not be valid, in which case this save doesn't do anything.
+            work.save
+            Hyrax.config.callback.run(:after_create_fileset, file_set, user)
+          end
+          work
```

Comment:
This is the only change to the original attach_to_work.  It now returns the updated work.